website/static/js/institutionSignIn.js

```diff
@@ -18,11 +18,26 @@ var ViewModel = function() {
                 isCors: true
             }
         ).done(function (response) {
-            self.instNames(response.data.map(function(item){
-                var name = item.attributes.name;
-                self.insts[name] = item.attributes.auth_url + '&target=' + encodeURIComponent(window.contextVars.institution_redirect);
-                return name;
-            }));
+            self.instNames(
+                response.data.map(function(item){
+                    var name = item.attributes.name;
+                    self.insts[name] = item.attributes.auth_url + '&target=' + encodeURIComponent(window.contextVars.institution_redirect);
+                    self.insts[item.id] = name;
+                    if (item.attributes.auth_url){
+                        return name;
+                    }
+                }).filter(function (item) {
+                   return typeof item !== 'undefined';
```

Comment:
Response doesn't. Not returning anything on ` .map ` (which happens if the item does not have an ` auth_url `) adds an undefined object to the list. This is here to clean up those undefineds.